tools/devtool

```diff
@@ -1082,69 +1150,134 @@ cmd_prepare_release() {
 #
-cmd_tag() {
+cmd_release() {
+    local user="dianpopa"
+    local prerelease="false"
+    local repo="firecracker"
+    local remote="origin"
 
     # Parse any command line args.
     while [ $# -gt 0 ]; do
         case "$1" in
-            "-h"|"--help")      { cmd_help; exit 1;    } ;;
-            *)                  { version="$1"; break; } ;;
+            "-h"|"--help")      { cmd_help; exit 1; } ;;
+            "-v"|"--version")
+                shift
+                local version="$1"
+                ;;
+            "-t"|"--token")
+                shift
+                local token="$1"
+                ;;
+             "-u"|"--username")
+                shift
+                user="$1"
+                ;;
+              "-r"|"--repo")
+                shift
+                repo="$1"
+                ;;
+              "-m"|"--remote")
+                shift
+                remote="$1"
+                ;;
+              "-p"|"--prerelease")
+                shift
+                prerelease="true"
+                ;;
+            *)
+                die "Unknown argument: $1. Please use --help for help."
+            ;;
         esac
         shift
     done
 
     validate_version "$version"
 
-    declare pat_release="^## \[([0-9]+\.){2}[0-9]+\]"
-    declare changelog="$FC_ROOT_DIR/CHANGELOG.md"
-
-    grep -q "\[$version\]" "$changelog"
-    ok_or_die "No changelog entry for release $version can be found."
-
-    # We work with the assumption that the changelog has already been updated
-    # and contains a header (and a corresponding section) for the new release.
-
-    # Step 1: Get all release numbers.
-    all_releases=($(grep -E "$pat_release" "$changelog"))
+    # Obtain the branch this script is run from.
+    branch=$(get_branch)
+    say "Will start creating a draft release for branch $branch ..."
+    get_user_confirmation || die "Aborted."
 
-    # Step 2: Trim out headers (`##`).
-    all_releases=(${all_releases[@]//##*})
+    # Check that the release tag associated with the provided version
+    # does not already exist. If the tag exists, we will not try to
+    # post a release on GitHub.
+    check_release_tag "$version" || skip_github_release=true
+
+    if [[ -z "$skip_github_release" ]]
+    then
+      if [[ -z "$token" ]]
+      then
+          say_err "In order to draft a GitHub release you need to pass a Github API Access Token!"
+      else
+          # Start by creating a local tag and associate to it a description.
+          say "Creating local tag..."
+          create_local_tag "$version" || die "Could not create local tag v"$version"."
+
+          # From now on, any function that will fail will also need to revert creation of local tag.
+          say "Continue with remote tag push?"
+          get_user_confirmation || die "Local tag not pushed to remote "$remote"."
+          push_local_tag "$version" "$remote" || delete_local_tag "$version" "Failed to push local tag v"$version"."
```

Comment:
I think we should stop when encountering any of these errors (do `die` after each of the following `delete_local_tag/delete_remote_tag`)

Reply:
Nevermind, on a second look I see that `die` is called inside `delete_local/remote_tag`. Resolving this.